src/style/style.js

```diff
@@ -668,7 +673,7 @@ class Style extends Evented {
         return true;
     }
 
-    addImage(id: string, image: StyleImage) {
+    addImage(id: string, image: StyleImage): ?this {
```

Comment:
Nit: `?this` as a return type exposes an API mistake — to make chaining work consistently, we should return `this` regardless of what happens inside these methods. So, maybe worth adding `return this` at the end and changing type to non-optional `this` in spots like this?